app/helpers/application_helper.rb

```diff
@@ -249,9 +249,7 @@ def author_links(post, linked: true, colored: false)
   end
 
   def allowed_boards(obj, user)
-    valid_ids = BoardAuthor.where(user: user).pluck(:board_id)
-    valid_ids << obj.board_id if obj.board_id.present?
-    have_authors = BoardAuthor.select(:board_id).distinct.pluck(:board_id)
-    Board.where(id: valid_ids).or(Board.where(creator: current_user)).or(Board.where.not(id: have_authors)).ordered
+    authored_ids = BoardAuthor.where(user: user).pluck(:board_id)
```

Comment:
I think this can use `.select(:board_id)` and then it'll end up being used as a subquery in the next query. This doesn't change much, though, so we can leave as is if you want!

Reply:
Which way does the performance effect go on that?

Reply:
Deferring to the database is ~always faster than doing things in our own code. It probably isn't a huge difference, though, because this will be a relatively tiny list, but loading it into our own code to generate another query just adds another round trip to the database.

Reply:
So, like, if we were checking it more than one place, we might want to load it into code so the database doesn't do it five times, but under these circumstances...

Reply:
Yeah, exactly!